### ISSUE TITLE
Donor wall is broken

### ISSUE THREAD

**Issue by yol** [xbmc/kodi-tv#190]
This doesn't look right: https://kodi.tv/donate/wall/

**Comment by yol** [comment]
> I don't have access to the AWS database anymore

You have now :)

PayPal: The entries might be from stray PayPal IPNs that arrived late after changing the settings, or from recurring donations that were initially set up like this and now repeat those values every payment. I think we can fix up those that are there now manually, but if more of those crop up and it becomes a steady problem, I'll find a way to fix the values before they enter the DB.

Stripe: All transactions from Stripe came up empty as far as names are concerned. I checked the raw webhooks and `client_reference_id` is set values such as `"undefined"` or `"nullnull"` (as string). Could this be a problem with the form on the website?

**Comment by pkscout** [comment]
I'll check this weekend. Of course this all worked it staging, so not sure yet if it's just a weird transition issue or something else.

**Comment by yol** [comment]
I've just taken a very quick look, my feeling is that there might be a confusion with `otForum` vs. `forum` variables in the JS

**Comment by pkscout** [comment]
@yol was right. I had changed some variable names at some point and missed a couple references to otForum and then apparently didn't test things again. Bad me. The form is fixed now and was tested locally and in staging. I also got into the AWS DynamoDB and deleted any of the bad references (the null names as well as the "mention my name" ones) and rebuilt the site. Those donations now show an Anonymous. I **think** that resolves all the issues, but I'll leave this open while we keep an eye on it and confirm in production.

**Comment by yol** [comment]
The PayPal issues are still there and I think that "old" donation subscriptions (i.e., that were made on the old site) will just keep their option_values indefinitely. So I've updated the aws scripts to fix them up automatically, see https://github.com/xbmc/aws/commit/e70a9513282a0d23543f6e351b418ebf18ef8198

@Darren-Hill has mentioned that he still receives IPN failure notifications, so I'm going to look into that too once I can log into PayPal. It seems like many are getting successfully delivered to the new address, though.

For the Stripe side of things, I think the fix got rid of the `undefined` stuff, but we still get `nullnull` quite often. I think #199 should fix that.

I also went through the DynamoDB again and removed all the `null` entries and fixed the wrong PayPal entries manually.
